### PR TITLE
Update coverage docs

### PR DESCRIPTION
**Pull request by richkadel**
r? @tmandry 

Before landing this PR, we should wait for:

* rust-lang/rust#84815 - because this new version of the dev guide document refers to a revised path to the Rust Unstable Book page on the compiler flag `-Z instrument-coverage`
* rust-lang/rust#84797 - because this new version of the document refers to coverage tests in `src/test/run-make`, the new location as of that PR

### CODE REVIEW

File: src/llvm-coverage-instrumentation.md
Comment (jyn514) [inline src/llvm-coverage-instrumentation.md:36]: This link is a 404. https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/instrument-coverage.html

Why did you change it? The old link works fine.

File: src/llvm-coverage-instrumentation.md
Comment (jyn514) [inline src/llvm-coverage-instrumentation.md:86]: Why change this?

File: src/llvm-coverage-instrumentation.md
Comment (jyn514) [inline src/llvm-coverage-instrumentation.md:115]: Why change this?

File: src/llvm-coverage-instrumentation.md
Comment (jyn514) [inline src/llvm-coverage-instrumentation.md:154]: Do you happen to have a link to the PR changing this? Has it already landed or is the PR still open?

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:36]: We'll need to land this after landing rust-lang/rust#84815

I'll convert this PR to "draft" until then.

When I reviewed that documentation, I realized the page name was non-standard. (I'm not sure why we didn't catch it before.) The page is supposed to match the compiler flag name.

Some external links still point to the old name, so I just converted that old page to include a link to the new one. But for the documentation we control, we should use the correct link (once it's active).

https://github.com/rust-lang/rust/pull/84815/files#diff-766876fc4110b68305d45c7c034b471975796591292f859ae61a6a5d1caf361b

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:86]: Sorry. I didn't change this intentionally. My editor seems to have aggressive re-formatting options turned on, and I assumed these were from some Rust-specific settings (like how the editor picks up rust formatting from the committed `rustfmt.toml`).

I'll revert any unnecessary changes.

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:115]: Not intended, as explained above. Sorry.

File: src/llvm-coverage-instrumentation.md
Comment (jyn514) [reply src/llvm-coverage-instrumentation.md:86]: No worries, thanks for taking care of it :)

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:154]: Yes, this landed on March 17. @tmiasko added the ability to support inlining with coverage.

rust-lang/rust#83080

See: https://github.com/rust-lang/rust/pull/83080/files#diff-f10b4e4dbb6fc3d26ea86854c45386cf59c2771c75a923f694b4c174e03e0970R114-R115

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:86]: Just FYI, I reverted the unintended changes made by the formatter, but I also compared the style in this doc to the style in the rest of the Rustc Dev Guide.

This doc was not internally consistent (italics/emphasis used both `_word_` and `*word*` with the same result, and bulleted items used both `*` and `-`. I chose the most common usage in the dev guide and make this document consistent (`_` and `-`, respectively).

Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:86]: Oh, and I rewrapped lines that were too long, compared to a standard line length used in most of the document.

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:115]: Leaving this one as-is, to be internally consistent (see my other comment explaining what I kept and why).

File: src/llvm-coverage-instrumentation.md
Comment (tmandry) [inline src/llvm-coverage-instrumentation.md:189]: this reads a bit funny :)

File: src/llvm-coverage-instrumentation.md
Comment (tmandry) [inline src/llvm-coverage-instrumentation.md:288]: This should change to `run-make` if that change lands

**Comment by tmandry** [comment]
Left a couple nits, otherwise LGTM

@bors delegate+

**Comment by camelid** [comment]
FYI, bors isn't enabled for rustc-dev-guide; use GitHub-native approvals instead :)

**Comment by tmandry** [comment]
Ah, thanks :)

**Review by tmandry — APPROVED** [review]

**Comment by richkadel** [comment]
Thanks @tmandry !

I don't expect any changes, but leaving it as a WIP / Draft until the changes to Rust Unstable Book land, which should resolve the 404.

Until then, we don't want to merge this.

**Comment by richkadel** [comment]
(I will address the review comments though... just to be clear. Thanks!)

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md]: Ha! You're right. I just removed the line. It's not really necessary.

File: src/llvm-coverage-instrumentation.md
Comment (richkadel) [reply src/llvm-coverage-instrumentation.md:287]: I changed it to `run-make` originally, and then we had the CI issues, so I've changed it back. I don't know how soon that will be resolved, so I think we should land this change now, and update this when it actually does move.

**Comment by camelid** [comment]
> Before landing this PR, we should wait for:
> 
>     * [rust-lang/rust#84815](https://github.com/rust-lang/rust/pull/84815) - because this new version of the dev guide document refers to a revised path to the Rust Unstable Book page on the compiler flag `-Z instrument-coverage`
> 
>     * [rust-lang/rust#84797](https://github.com/rust-lang/rust/pull/84797) - because this new version of the document refers to coverage tests in `src/test/run-make`, the new location as of that PR

Looks like these two have been merged! @rustbot label: -blocked +waiting-on-review

**Comment by richkadel** [comment]
@camelid - Thanks. It has been reviewed and approved, and is ready to be merged (unless I misunderstand the label you added).

**Pull request merged**
